modules/dists/ReplicatedDist.chpl

```diff
@@ -410,31 +366,19 @@ proc ReplicatedDom.dsiGetIndices(): rank * range(idxType,
 }
 
 // Iterators over the domain's indices (serial, leader, follower).
-// Our semantics: yield each of the domain's indices once per each locale.
+// Our semantics: yield each of the local domain's indices.
 
-// Serial iterator: the compiler forces it to be completely serial
+// Serial iterator
 iter ReplicatedDom.these() {
-  // compiler does not allow 'on' here (see r16137 and nestedForall*)
-  // so instead of ...
-  //---
-  //for locDom in localDoms do
-  //  on locDom do
-  //    for i in locDom.domLocalRep do
-  //      yield i;
-  //---
-  // ... so we simply do the same a few times
   var dom = redirectee();
-  for count in 1..#numReplicands do
-    for i in dom do
-      yield i;
+  for i in dom do
+    yield i;
 }
 
 iter ReplicatedDom.these(param tag: iterKind) where tag == iterKind.leader {
-  coforall locDom in localDoms do
-    on locDom do
-      // there, for simplicity, redirect to DefaultRectangular's leader
-      for follow in locDom.domLocalRep._value.these(tag) do
-        yield follow;
+  // for simplicity, redirect to DefaultRectangular's leader
```

Comment:
"for simplicity" implies that we have other choices, which practically speaking we don't.
Let's remove "for simplicity".